app/src/org/commcare/activities/CommCareHomeActivity.java

```diff
@@ -1098,6 +1098,10 @@ private boolean checkAndStartUnsentFormsTask(final boolean syncAfterwards,
     @Override
     protected void onResume() {
         super.onResume();
+        if (redirectedToLogin) {
+            finish();
+            return;
```

Comment:
Or rather, make `SessionAwareCommCareActivity.onResume` only call `onResumeSessionSafe` if the session hasn't expired, otherwise redirecting to login.